apis/air_quality/src/lib.rs

```diff
@@ -0,0 +1,92 @@
+#![no_std]
+
+use core::cell::Cell;
+use libtock_platform::{
+    share::scope, share::Handle, DefaultConfig, ErrorCode, Subscribe, Syscalls,
+};
+use ReadType::{ReadCO2, ReadTVOC};
+
+#[cfg(test)]
+mod tests;
+
+enum ReadType {
+    ReadCO2 = READ_CO2 as isize,
+    ReadTVOC = READ_TVOC as isize,
+}
+
+pub struct AirQuality<S: Syscalls>(S);
+
+impl<S: Syscalls> AirQuality<S> {
+    pub fn exists() -> Result<(), ErrorCode> {
+        S::command(DRIVER_NUM, EXISTS, 0, 0).to_result()
+    }
+
+    pub fn register_listener<'share>(
+        listener: &'share Cell<Option<(u32,)>>,
+        subscribe: Handle<Subscribe<'share, S, DRIVER_NUM, 0>>,
+    ) -> Result<(), ErrorCode> {
+        S::subscribe::<_, _, DefaultConfig, DRIVER_NUM, 0>(subscribe, listener)
+    }
+
+    pub fn unregister_listener() {
+        S::unsubscribe(DRIVER_NUM, 0)
+    }
+
+    pub fn read_co2() -> Result<(), ErrorCode> {
+        S::command(DRIVER_NUM, READ_CO2, 0, 0).to_result()
+    }
+
+    pub fn read_tvoc() -> Result<(), ErrorCode> {
+        S::command(DRIVER_NUM, READ_TVOC, 0, 0).to_result()
+    }
+
+    pub fn read_co2_sync() -> Result<u32, ErrorCode> {
+        Self::read_data_sync(ReadCO2)
+    }
+
+    pub fn read_tvoc_sync() -> Result<u32, ErrorCode> {
+        Self::read_data_sync(ReadTVOC)
+    }
+
+    fn read_data_sync(read_type: ReadType) -> Result<u32, ErrorCode> {
```

Comment:
This is not similar to the async API, either make two functions either make one function for the synchronous API.

Reply:
As stated before, this function is not public, it has two wrappers that are public so the user of the API doesn't need to match the enum value for each command.